### PR TITLE
Update `CircularAperture.plot()` to return patches

### PR DESCRIPTION
**Pull request by b1quint**
I updated the `photutils.aperture.core.PixelAperture.plot` method so it returns the Patches to be used later. For example, I wanted to compare the apertures found with `DAOStarFind` and `IRAFStarFind` and add a legend. Having the patches allows me to create a legend to identify the apertures. 

Here is an example of what I wanted to do:
dao_vs_iraf

And here is the code that generated the code:
```python
...
iraf_patches = iraf_apertures.plot(color='blue', lw=1.5, alpha=0.5, axes=ax)
dao_patches = dao_apertures.plot(color='red', lw='1.5', alpha=0.5, axes=ax)

ax.legend([iraf_patches[0], dao_patches[0]], ['IRAFStarFind', 'DAOStarFind'],
         framealpha=1)
...
```

I added some tests to `photutils.aperture.tests/test_circle.py` and updated the docstring of the updated function. Please, let me know if you need anything else here.

### CODE REVIEW

File: photutils/aperture/tests/test_circle.py [hunk 1]
```diff
@@ -6,29 +6,76 @@
 from astropy.coordinates import SkyCoord
 import astropy.units as u
 import numpy as np
+import pytest
 
 from .test_aperture_common import BaseTestAperture
-from ..circle import (CircularAperture, CircularAnnulus, SkyCircularAperture,
-                      SkyCircularAnnulus)
+from ..circle import (
```
Comment (bsipocz) [inline photutils/aperture/tests/test_circle.py]: please remove the code reformatting commit when rebasing

Comment (larrybradley) [reply photutils/aperture/tests/test_circle.py]: Agreed.

**Comment by b1quint** [comment]
I can try to do the rebase myself. That would be a good exercise. But I did not find (yet) a documentation on how to do it. If you have it, would you, please, share it?

**Comment by b1quint** [comment]
I performed the rebase as described in the [AstroPy Contributing Documentation](http://docs.astropy.org/en/stable/development/workflow/development_workflow.html#how-to-rebase) but there is one CI build that is still not passing.

**Comment by b1quint** [comment]
I have been trying to update the change log to be approved by the `astrobot` but I cannot see what is the problem. Would it be possible to have a bit of more information to prevent me doing several commits?

**Comment by bsipocz** [comment]
> I have been trying to update the change log to be approved by the astrobot but I cannot see what is the problem. Would it be possible to have a bit of more information to prevent me doing several commits?

Please ignore the bot for now. v0.7 has been released a couple of days ago, and the bot complains that the changelog entry for this PR is in that section rather than the new v0.8. However the content needs to be discussed first, so I would suggest to wait for @larrybradley to do it before trying to satisfy the bot.

**Comment by b1quint** [comment]
Hey all! 

I am sorry I left this PR unattended. I would like to finish it and I was wondering if it would be best to merge `master` into it or if you prefer that I use a git rebase.

**Comment by bsipocz** [comment]
A rebase is always preferred, thanks for asking!

**Comment by b1quint** [comment]
I already performed the rebase. Now the two astropy-bot checks are the only ones that do not pass. What should I do to meet their requirements? Is there any documentation so I can update this PR?

**Comment by bsipocz** [comment]
If you move the changelog entry to the top for the next release (1.1), then the bot should pass, too.

**Comment by larrybradley** [comment]
Thanks, @b1quint.  I will look at this later today.  I need to go back and review https://github.com/astropy/regions/issues/296 to refresh my memory.

File: CHANGES.rst [hunk 2]
```diff
@@ -1,7 +1,9 @@
 1.1.0 (unreleased)
 ------------------
 
-- No changes yet
+- ``photutils.aperture.core.PixelAperture.plot``
```
Comment (larrybradley) [inline CHANGES.rst:4]:
```suggestion
- ``photutils.aperture``
```

File: CHANGES.rst [hunk 3]
```diff
@@ -1,7 +1,9 @@
...
+
+  - Now returns a list of ``matplotlib.patches.Patch``. [#923]
```
Comment (larrybradley) [inline CHANGES.rst:6]:
```suggestion
  - The ``PixelAperture.plot()`` method now returns a list of 
    ``matplotlib.patches.Patch`` objects. [#923]
```

File: CHANGES.rst [hunk 4]
```diff
@@ -262,6 +264,10 @@ New Features
     methods was removed to implement the shared WCS interface.  All
     WCS transforms now include distortions (if present). [#899]
 
+  - ``astropy.aperture.core.PixelAperture.plot`` now returns the patches that
+    are added to the new/existing axes. That allows manipulating them
+    individually or adding them to a legend. [#923]
+
```
Comment (larrybradley) [inline CHANGES.rst:270]:
```suggestion
```

Version 1.0.1 has already been released.

**Review by larrybradley — CHANGES_REQUESTED** [review]
Thanks, @b1quint.  Please remove the changelog entry under 1.0.1 (it's already released).  I made some formatting changes for the 1.1 changelog entry -- you can simply accept my suggestions.

Also, one can include a `label` in the `plot()` call instead of passing it to `legend`, like this:

```python
from photutils import CircularAperture

data = np.ones((50, 50))
ap1 = CircularAperture(((10, 20), (30, 40)), r=3)
ap2 = CircularAperture(((10, 20), (30, 40)), r=5)
ap3 = CircularAperture((30, 20), r=5)

fig, ax = plt.subplots(figsize=(5, 5))
ax.imshow(data, cmap='Greys')
ap1_patches = ap1.plot(ax, color='red', label='Aper 1')
ap2_patches = ap2.plot(ax, color='blue', label='Aper 2')
ap3_patches = ap3.plot(ax, color='green', label='Aper 3')
handles = (ap1_patches[0], ap2_patches[0], ap3_patches[0])
ax.legend(handles=handles)
```

**Comment by b1quint** [comment]
@larrybradley When you say "one can include a `label` in the `plot()` call instead of passing it to `legend`", it was not clear to me if you were just giving some example or you expected me to change anything. I tested the piece of code you wrote, confirmed it works as you suggested. Maybe complement the docstrings with it?

**Review by larrybradley — APPROVED** [review]
Thanks, @b1quint.  I was just pointing out another way to set the labels.  There's no need to re-document everything from matplotlib here, but I may add an example plot.

**Pull request merged**